### PR TITLE
feat: add `Expr` methods: `as_tuple`, `as_parenthesized`, `as_index`, `as_if`

### PR DESCRIPTION
**Pull request by asterite**
# Description

## Problem

Part of #5668

## Summary

Just four, to keep PRs small.

## Documentation

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [x] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs [hunk 1]
```diff
@@ -48,6 +48,10 @@ impl<'local, 'context> Interpreter<'local, 'context> {
             "array_len" => array_len(interner, arguments, location),
             "as_slice" => as_slice(interner, arguments, location),
             "expr_as_function_call" => expr_as_function_call(arguments, return_type, location),
+            "expr_as_if" => expr_as_if(arguments, return_type, location),
+            "expr_as_index" => expr_as_index(arguments, return_type, location),
+            "expr_as_parenthesized" => expr_as_parenthesized(arguments, return_type, location),
```
Comment (jfecher) [inline compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs:53]: I don't think we should expose Ast::Parenthesized, I see it more of an internal detail to preserve formatting.
Instead we should probably have a check for it in `expr_as` to automatically recur and match on the expression inside of it instead.

**Review by jfecher — CHANGES_REQUESTED** [review]
> [For Experimental Features] Documentation to be submitted in a separate PR.

Can you select this option for documention for these PRs? We'll want to document these in the future soon.

File: compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs (hunk 1 above)
Comment (asterite) [reply compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs:53]: Ah, I also thought about this and I wasn't sure. I'll do it 👍

**Review by jfecher — APPROVED** [review]
:+1:

**Pull request merged**